src/mpi/attr/attrutil.c

```diff
@@ -216,14 +208,19 @@ int MPIR_Attr_dup_list(int handle, MPIR_Attribute * old_attrs, MPIR_Attribute **
   fn_exit:
     return mpi_errno;
   fn_fail:
+    /* In case of any failure, remove the attributes duplicated so far
```

Comment:
This one is important. If during the duplication of an attribute list, any of the user-define copy_fn functions fail, then I attempt to deallocate the list duplicated so far. Note however that if any delete_attr_fn function fails, I just ignore that error, IMHO reporting the one from copy_fn should take precedence.